src/utils/type-guards.ts

```
@@ -0,0 +1,24 @@
+  return Object.prototype.hasOwnProperty.call(obj, key);
+}
+
+/**
+ * Type guard to check if the value is of the expected type
+ *
+ * @param value The value to check
+ * @param type The expected type ('string', 'number', 'boolean', etc.)
+ * @returns True if the value is of the expected type, false otherwise
+ */
+export function isType<T>(value: unknown, type: string): value is T {
+  return typeof value === type;
```

Biome (1.9.4)

[error] Invalid `typeof` comparison value: this expression is not a string literal

not a string literal

(lint/suspicious/useValidTypeof)